pkgs/development/python-modules/transmission-rpc/typing-extensions.patch

```diff
@@ -0,0 +1,103 @@
+diff --git a/poetry.lock b/poetry.lock
+index 79eb8a3..f726edb 100644
+--- a/poetry.lock
++++ b/poetry.lock
+@@ -745,11 +745,11 @@ python-versions = "*"
+ 
+ [[package]]
+ name = "typing-extensions"
+-version = "3.10.0.2"
+-description = "Backported and Experimental Type Hints for Python 3.5+"
++version = "4.0.1"
++description = "Backported and Experimental Type Hints for Python 3.6+"
```

Comment:
Why do we need to modify version, description and python-versions? Isn't that better suited for a pullrequest against the upstream?

Reply:
Raw output after invoking `poetry update typing_extensions`. Didn't want to mess anymore with it.
This is a temp fix, until they tag the version they have upstream with the correct version :)

Reply:
Instead of a patch we could do a more minimal approach of using `substituteInPlace` to relax the version constraint and update the version in `pyproject.toml`.

```
  postPatch = ''
    substituteInPlace pyproject.toml \
      --replace 'version = "3.10.0.2"' 'version = "4.0.1"' \
      --replace 'typing_extensions = ">=3.7.4.2,<4.0.0.0"' 'typing_extensions = "*"'
  '';
```

Reply:
Works for me! Changed in latest version :)

Reply:
Great, then let's drop the patch.